ambry-replication/src/test/java/com.github.ambry.replication/ReplicationTest.java

```diff
@@ -807,6 +809,51 @@ public void replicaThreadTest() throws Exception {
     // 1 expired + 1 corrupt + 1 put (never present) + 1 deleted (never present) expected missing buffers
     verifyNoMoreMissingKeysAndExpectedMissingBufferCount(remoteHost, localHost, replicaThread, replicasToReplicate,
         idsToBeIgnoredByPartition, storeKeyConverter, expectedIndex, expectedIndex + 1, 4);
+
+    // tests to verify replica thread throttling function in the following steps:
+    // 1. all replicas are in sync, thread level sleep and replica quarantine are both enabled.
+    // 2. add put messages to some replica and verify that replication for replicas remain disabled.
+    // 3. forward the time so replication for replicas are re-enabled and check replication resumes.
+    // 4. add more put messages to ensure replication happens continuously when replication is needed.
+
+    // 1. verify that the replica thread sleeps and replicas are temporarily disable when all replicas are synced.
+    long currentTimeMs = time.milliseconds();
+    List<List<RemoteReplicaInfo>> replicasToReplicateList = new ArrayList<>(replicasToReplicate.values());
+    replicaThread.replicate(replicasToReplicateList);
+    // replicate is called twice to trigger the sleep (replication for all replica is disabled then replicate is put to sleep).
+    replicaThread.replicate(replicasToReplicateList);
+    assertTrue("Replicas are in sync, replica thread should sleep by replication.sleep.duration.ms",
+        currentTimeMs + config.replicationThreadSleepDurationMs <= time.milliseconds());
```

Comment:
this should be `==` right (or in other words, we should be able to use `assertEquals()`)? Nothing else should be advancing the time (this test is useless if that is happening)